linked_list.rb

```diff
@@ -19,77 +19,232 @@ def initialize
   # method to add a new node with the specific data value in the linked list
   # insert the new node at the beginning of the linked list
   def insert(value)
-    puts "Not implemented"
+    node = Node.new(value)
+    if @head == nil
+      @head = node
+    else
+      node.next = @head
+      @head = node
+    end
+
+    # puts "Not implemented"
   end
 
   # method to find if the linked list contains a node with specified value
   # returns true if found, false otherwise
   def search(value)
-    puts "Not implemented"
+    current = @head
+    while current != nil
+      if current.data == value
+        return true
+      else
+        current = current.next
+      end
+
+    end
+    return false
+
+    # puts "Not implemented"
   end
 
   # method to return the max value in the linked list
   # returns the data value and not the node
   def find_max
-    puts "Not implemented"
+
+    current = @head
+    max = current.data
+    while current != nil
+      if current.data > max
+        max = current.data
+      else
+        current = current.next
+      end
+    end
+    return max
+    # puts "Not implemented"
   end
 
   # method to return the min value in the linked list
   # returns the data value and not the node
   def find_min
-    puts "Not implemented"
+    current = @head
+    min = current.data
+    while current != nil
+      if current.data < min
+        min = current.data
+      else
+        current = current.next
+      end
+    end
+    return min
+    # puts "Not implemented"
   end
 
   # method that returns the length of the singly linked list
   def length
-    puts "Not implemented"
+    return 0 if @head == nil
+    current = @head
+    size = 0
+    while current != nil do
+      size += 1
+      current = current.next
+    end
+    return size
   end
 
   # method to return the value of the nth element from the beginning
   # assume indexing starts at 0 while counting to n
   def find_nth_from_beginning(n)
-    puts "Not implemented"
+    current = @head
+    size = 0
+    until size == n do
+      size += 1
+      current = current.next

```

Comment:
This will have an issue if n is greater than number of nodes in the linked list. Always check to ensure that @head is not nil. Then, in the loop, check to ensure that current is not nil before accessing current.next.